### PR TITLE
[S3] add sns web identity provider to s3 client

### PR DESCRIPTION
**Pull request by ZhiHanZ**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Use auto-refresh web identity token on kubernetes to access S3 bucket

## Changelog

- New Feature

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/3123

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
@mergify update

**Review by PsiACE — APPROVED** [review]
LGTM

File: common/dal/src/accessors/aws_s3/s3.rs [hunk 1]
```diff
@@ -53,15 +53,31 @@ impl S3 {
...
-                    ErrorCode::DALTransportError(format!(
-                        "failed to create default credentials provider, {}",
-                        e
-                    ))
-                })?,
-                dispatcher,
-            ),
+            None => {
+                // check on k8s admission webhook injection
+                if std::env::var("AWS_WEB_IDENTITY_TOKEN_FILE").is_ok() {
```
Comment (BohuTANG) [inline common/dal/src/accessors/aws_s3/s3.rs:58]: Why not add a new s3 storage config?

Comment (ZhiHanZ) [reply common/dal/src/accessors/aws_s3/s3.rs:58]: does not seems to be make sense... using SNS web identity means to use token generated by some controllers with fixed env variable in every instance, there seems have little space on self customization,

Comment (BohuTANG) [reply common/dal/src/accessors/aws_s3/s3.rs:58]: If we inline a env in the codes, it looks very tricky, and feels like a back door was left open, we should make this configuration public although we are using it from env

Comment (ZhiHanZ) [reply common/dal/src/accessors/aws_s3/s3.rs:58]: https://www.kubeflow.org/docs/distributions/aws/deploy/install-kubeflow/#option-1-use-aws-iam-roles-for-service-accounts-default-and-recommended

I would like to add similar feature flag to let user use IAM policies.

Comment (BohuTANG) [reply common/dal/src/accessors/aws_s3/s3.rs]: Nice, thank you @ZhiHanZ

**Pull request merged**